kable-core/src/jsMain/kotlin/Bluetooth.kt

```diff
@@ -1,103 +1,36 @@
-    } else {
-        val isAvailable = bluetooth.getAvailability().await()
-        if (isAvailable) Available else Unavailable(reason = null)
+/**
+ * @return [Bluetooth] object or `null` if bluetooth is [unavailable](https://developer.mozilla.org/en-US/docs/Web/API/Bluetooth#browser_compatibility).
+ */
+internal fun bluetoothOrNull(): Bluetooth? {
+    val navigator = try {
+        window.navigator
+    } catch (e: ReferenceError) {
+        // ReferenceError: window is not defined
+        return null
     }
-    emit(availability)
+    return navigator.bluetooth.takeIf { it !== undefined }
 }
 
-// Deliberately NOT cast `as Bluetooth` to avoid potential class name collisions.
-@Suppress("UnsafeCastFromDynamic")
-internal val bluetooth: JsBluetooth
-    get() = checkNotNull(safeWebBluetooth) { "Bluetooth unavailable" }
-
-// In a node build environment (e.g. unit test) there is no window, guard for that to avoid build errors.
-private val safeWebBluetooth: dynamic =
-    js("typeof(window) !== 'undefined' && window.navigator.bluetooth")
-
-public fun CoroutineScope.requestPeripheral(
-    options: Options,
-    builderAction: PeripheralBuilderAction = {},
-): Promise<Peripheral> = bluetooth
-    .requestDevice(options.toRequestDeviceOptions())
-    .then { device -> peripheral(device, builderAction) }
-
 /**
- * Convert public API type to external Web Bluetooth (JavaScript) type.
- *
- * According to the `requestDevice`
- * [example](https://developer.mozilla.org/en-US/docs/Web/API/Bluetooth/requestDevice#example), the form of the
- * JavaScript object should be similar to:
- * ```
- * let options = {
- *   filters: [
- *     {services: ['heart_rate']},
- *     {services: [0x1802, 0x1803]},
- *     {services: ['c48e6067-5295-48d3-8d5c-0395f61792b1']},
- *     {name: 'ExampleName'},
- *     {namePrefix: 'Prefix'}
- *   ],
- *   optionalServices: ['battery_service']
- * }
- * ```
+ * @throws IllegalStateException If bluetooth is [unavailable](https://developer.mozilla.org/en-US/docs/Web/API/Bluetooth#browser_compatibility).
  */
-private fun Options.toRequestDeviceOptions(): RequestDeviceOptions = jso {
-    val jsFilters = this@toRequestDeviceOptions.filters.toBluetoothLEScanFilterInit()
-    if (jsFilters.isNotEmpty()) {
-        filters = jsFilters.toTypedArray()
-    } else {
-        acceptAllDevices = true
+internal fun bluetoothOrThrow(): Bluetooth {
+    val navigator = try {
+        window.navigator
+    } catch (e: ReferenceError) {
+        // ReferenceError: window is not defined
+        throw IllegalStateException("Bluetooth unavailable", e)
     }
-    if (!this@toRequestDeviceOptions.optionalServices.isNullOrEmpty()) {
-        optionalServices = this@toRequestDeviceOptions.optionalServices
-            .map(Uuid::toBluetoothServiceUUID)
-            .toTypedArray()
+    val bluetooth = navigator.bluetooth
+    if (bluetooth === undefined) {
+        error("Bluetooth unavailable")
```

Comment:
This seems duplicative - we have the same logic as above, and both `throw IllegalStateException("Bluetooth unavailable", e)` and `error("Bluetooth unavailable")` here. How about we just write what the function name suggests?
```kotlin
internal fun bluetoothOrThrow(): Bluetooth =
    bluetoothOrNull() ?: error("Bluetooth unavailable")
```

Reply:
Oh, just realized the `throw` is re-wrapping the exception. Derp. 🫠 Carry on...